src/components/BottomBar/Missions/Timeline.jsx

```diff
@@ -0,0 +1,460 @@
+    const paddingY = padding / k; // Make sure padding doesn't get stretched when zooming
+    const radiusY = radiusPhase / k; // Same here
+    return (
+      <rect
+        key={`${phase.name}${startTime.toString()}${endTime.toString()}${color}`}
+        x={xScale(nestedLevels - nestedLevel - 1) - padding}
+        y={yScale(endTime) - (paddingY)}
+        ry={radiusY}
+        rx={radiusPhase}
+        height={yScale(startTime) - yScale(endTime) + (2 * paddingY)}
+        width={xScale(1) - xScale(0) + (2 * padding)}
+        onClick={(e) => {
+          setDisplayedPhase({ type: DisplayType.phase, data: phase });
+          onClick(e, startTime);
+        }}
+        onMouseOver={(e) => mouseOver(e, "Phase", phase.name)}
+        onMouseLeave={mouseLeave}
+        className={isCurrent ? styles.barHighlighted : styles.bar}
+        strokeWidth={0}
+        style={color ? { fill: color, opacity: 1.0 } : null} // Override stylesheet
+      />
+    );
+  }
+
+  // Used for the current time indicator
+  function createLine(time, color, ref) {
+    // Check so time is valid
+    if (!(time instanceof Date && !isNaN(time))) {
+      return null;
+    }
+    const lineWidthScaled = lineWidth / k; // Ensure line doesn't get stretched when zooming
+    const yPosition = yScale(time) - (lineWidthScaled * 0.5); // Center line around time
+    return (
+      <rect
+        key={time.toUTCString()}
+        ref={el => ref ? ref.current = el : null}
+        x={margin.left}
+        y={yPosition}
+        height={lineWidthScaled}
+        width={width - margin.left - margin.right}
+        fill={color}
+        stroke={borderColor}
+        strokeWidth={borderWidth/k}
+      />
+    )
+  }
+
+  // Used for instrument activity / capture times
+  function createCircle(time, color, index) {
+    return (
+      <ellipse 
+        key={`${time?.toISOString()}${index}`}
+        cx={margin.left}
+        cy={yScale(time)}
+        ry={circleRadius / k}
+        rx={circleRadius}
+        fill={color}
+        onClick={(e) => onClick(e, time)}
+        className={styles.capture}
+        onMouseOver={(e) => mouseOver(e, "Instrument Activity")}
+        onMouseLeave={mouseLeave}
+      />
+    )
+  }
+
+  // Used for milestones
+  function createPolygon(date, color = undefined, noBorder = false, padding = 0) {
+    const time = makeUtcDate(date.date);
+    const width = polygonSize + (2 * padding);
+    const y = yScale(time) - (width * 0.5 / k);
+    const x = margin.left - width;
+    const centerOffsetX = 0.5 * width;
+    return (
+      <polygon
+        points={`0,${width * 0.5} ${width * 0.5},${width} ${width},${width * 0.5} ${width * 0.5},0`}
+        transform={`translate(${x + centerOffsetX}, ${y})scale(1, ${1/k})`}
+        fill={color}
+        stroke={borderColor}
+        strokeWidth={noBorder ? 0 : borderWidth}
+        key={`${time.toUTCString()}${padding}${color}`}
```

Comment:
Pretty complicated for just being a unique key :) 


How about making it a variable and add a comment why all these things are needed?